src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -272,9 +272,9 @@ public boolean canProvision(Label label) {
     // item in the list of configured Mesos labels.
     // TODO(vinod): The framework may not have the resources necessary
     // to start a task when it comes time to launch the slave.
-    if (label != null && slaveInfos != null) {
+    if (slaveInfos != null) {
       for (MesosSlaveInfo slaveInfo : slaveInfos) {
-        if (label.matches(Label.parse(slaveInfo.getLabelString()))) {
+        if (slaveInfo.matchesLabel(label)) {
```

Comment:
yes, this PR doesn't change the default behavior, if job doesn't have a label it will only run in slaves that don't have a label. By default mesos slaves will still use the `mesos` label